### PR TITLE
Update deploy without resource configuration

### PR DESCRIPTION
**Pull request by Flanker32**
- Get artifact from target folder when no resource specified 
- Throw exception when there are multi artifacts could be deployed

### CODE REVIEW

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java [hunk 1]
```diff
@@ -100,34 +107,27 @@ public static int convertSizeStringToNumber(String memory) throws MojoExecutionE
...
-
     public static String generateTimestamp() {
         return new SimpleDateFormat(DATE_FORMAT).format(new Date());
     }
 
-    public static File getArtifactFromConfiguration(SpringConfiguration springConfiguration) {
+    public static File getArtifactFromTargetFolder(MavenProject project) throws MojoExecutionException {
+        try {
+            final String targetFolder = project.getBuild().getDirectory();
+            final List<File> files = Files.walk(Paths.get(targetFolder))
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#listFiles(java.io.File,%20java.lang.String[],%20boolean)

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java [hunk 2]
```diff
@@ -176,17 +210,16 @@ public static boolean isExecutableJar(File file) {
...
-        } catch (Exception e) {
-            wrappedFuture.cancel(true);
-            return null;
-        } finally {
-            executorService.shutdown();
+    private static File getRunnableArtifactFromFiles(List<File> files) throws MojoExecutionException {
+        final List<File> runnableArtifacts = files.stream().filter(Utils::isExecutableJar).collect(Collectors.toList());
+        if (runnableArtifacts.size() == 1) {
+            return runnableArtifacts.get(0);
+        } else if (runnableArtifacts.size() == 0) {
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: isEmpty

Comment (andxu) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: replace the if -> else if -> else if to if if if

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java [hunk 3]
```diff
@@ -176,17 +210,16 @@ public static boolean isExecutableJar(File file) {
...
+            throw new MojoExecutionException(ARTIFACT_NOT_SUPPORTED);
+        } else {
+            final List<String> artifactNameLists = runnableArtifacts.stream()
+                    .map(file -> file.getName()).collect(Collectors.toList());
+            throw new MojoExecutionException(String.format(MULTI_ARTIFACT, String.join(",", artifactNameLists)));
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: collect(Collectors.join())

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java (hunk 1 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: Chang to use FileUtils

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java (hunk 2 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: Fixed.

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java (hunk 3 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: Fixed.

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java [hunk 4]
```diff
@@ -157,6 +150,40 @@ public static boolean isExecutableJar(File file) {
...
+            executorService.shutdown();
+        }
+    }
+
+    private static List<File> getArtifacts(List<Resource> resources) {
+        final List<File> result = new ArrayList<>();
+        final DirectoryScanner directoryScanner = new DirectoryScanner();
+        for (final Resource resource : resources) {
+            if (resource.getIncludes() != null && resource.getIncludes().size() > 0) {
+                directoryScanner.setBasedir(resource.getDirectory());
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: I think the following utils are doing the same thing
 org.apache.commons.io.filefilter.FileFilterUtils;
 org.apache.commons.io.filefilter.IOFileFilter;
 org.apache.commons.io.filefilter.NameFileFilter;
 org.apache.commons.io.filefilter.WildcardFileFilter;

Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: This will be done in following pr.

**Pull request merged**